### PR TITLE
Refactor goto functions

### PR DESCRIPTION
**Pull request by mrkvon**
Goto functions are custom routes for express based on (not url).
They have been very repetitive.
We replace them with a single function which produces the goto middleware.

### CODE REVIEW

File: controllers/goto/goto.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    // we're in the right router. continue...
+    if (go) return next();
+
+    // wrong router. check another one.
+    next('route');
+
+  };
+};
```
Comment (agatatalita) [inline controllers/goto/goto.js]: Alternative function with less assignments not checking always whole paths array: 
  

```
return function (req, res, next) {
    // checks if every path is present in the request
    let go = paths.every(path => _.has(req,path));

    // checking validity with schema
    // for success, schema is either not provided, or request is valid against it
    if (go && typeof(schema) === 'string' ) { 
      go = ajv.validate(`go#/${schema}`, req);
    }

    if (go) {
      // we're in the right router. continue...
      return next();
    }

    // wrong router. check another one.
    next('route');

  };
```

File: controllers/goto/goto.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    let go = true;
+    paths.forEach(path => {
+      go = go && _.has(req, path);
+    });
+
+    // checking validity with schema
+    // for success, schema is either not provided, or request is valid against it
+    if (go && typeof(schema) === 'string') {
+      const isSchemaValid = ajv.validate(`go#/${schema}`, req);
+      go = go && isSchemaValid;
```
Comment (agatatalita) [inline controllers/goto/goto.js]: go is always true here
`go = isSchemaValid;`
is enought

File: controllers/goto/goto.js (hunk 1 above)
Comment (agatatalita) [inline controllers/goto/goto.js]: Anyways clear and simple.

Comment (mrkvon) [inline controllers/goto/goto.js]: Very nice. Will replace. :+1:

**Pull request merged**